compiler/damlc/lib/DA/Cli/Damlc/Packaging.hs

```diff
@@ -50,33 +50,31 @@ import qualified DA.Pretty
 import SdkVersion
 
 -- | Create the project package database containing the given dar packages.
+--
+-- We differentiate between two kinds of dependencies.
+--
+-- deps (depencies in daml.yaml):
+--   This is intended for packages that include interface files and can
+--   be used directly.
+--   These packages have to be built with the same SDK version.
+--
+-- data-deps (data-dependencies in daml.yaml):
+--   This is intended for packages that have already been uploaded to the
+--   ledger. Based on the DAML-LF we generate dummy interface files
+--   and then remap references to those dummy packages to the original DAML-LF
+--   package id.
 createProjectPackageDb ::
        Options -> PackageSdkVersion -> [FilePath] -> [FilePath] -> IO ()
-createProjectPackageDb opts thisSdkVer deps0 dataDeps = do
-    let dbPath = projectPackageDatabase </> (lfVersionString $ optDamlLfVersion opts)
-    let
+createProjectPackageDb opts thisSdkVer deps dataDeps = do
+    let dbPath = projectPackageDatabase </> lfVersionString (optDamlLfVersion opts)
     -- Since we reinitialize the whole package db anyway,
     -- during `daml init`, we clear the package db before to avoid
     -- issues during SDk upgrades. Once we have a more clever mechanism than
     -- reinitializing everything, we probably want to change this.
     removePathForcibly dbPath
     createDirectoryIfMissing True $ dbPath </> "package.conf.d"
-    -- Expand SDK package dependencies using the SDK root path.
-    -- E.g. `daml-trigger` --> `$DAML_SDK/daml-libs/daml-trigger.dar`
-    -- Or, fail if not run from DAML assistant.
-    mbSdkPath <- handleIO (\_ -> pure Nothing) $ Just <$> getSdkPath
-    let isSdkPackage fp = takeExtension fp `notElem` [".dar", ".dalf"]
-        handleSdkPackages :: [FilePath] -> IO [FilePath]
-        handleSdkPackages =
-          let expand fp
-                | isSdkPackage fp
-                = case mbSdkPath of
-                    Just sdkPath -> pure $! sdkPath </> "daml-libs" </> fp <.> "dar"
-                    Nothing -> fail $ "Cannot resolve SDK dependency '" ++ fp ++ "'. Use daml-assistant."
-                | otherwise
-                = pure fp
-          in mapM expand
-    deps <- handleSdkPackages $ filter (`notElem` basePackages) deps0
+
+    deps <- expandSdkPackages (filter (`notElem` basePackages) deps)
     depsExtracted <- mapM extractDar deps
     let uniqSdkVersions = nubSort $ unPackageSdkVersion thisSdkVer : map edSdkVersions depsExtracted
     unless (length uniqSdkVersions <= 1) $
```

Comment:
Suggestion: instead of nubSorting, just compare all the deps' sdk versions to the current package sdk version. 

```suggestion
    let depsSdkVersions = map edSdkVersions depsExtracted
    unless (all (== unPackageSdkVersion thisSdkVer) depsSdkVersions) $
```